async_simple/Executor.h

```diff
@@ -91,12 +91,20 @@ class Executor {
     // `schedule` would return false if schedule failed, which means function
     // func will not be executed. In case schedule return true, the executor
     // should guarantee that the func would be executed.
-    virtual bool schedule(Func func) = 0;
+    //
+    // The priority_hint is used to guide the scheduling order, larger number
```

Comment:
Let's change the name to schedule_context and specify its type as uint64_t. And let's announce we (async_simple) only saves 16 bits and leave the other 48 bits for users. We can use only 4 bits to represent the priority.

And let's announce `YIELD_JOB` enum to avoid magic numbers.